pipeline/mutate/mutator_hydrator.go

```diff
@@ -172,33 +168,28 @@ func (a *MutatorHydrator) Mutate(r *http.Request, session *authn.AuthenticationS
 	}
 	req.Header.Set(contentTypeHeaderKey, contentTypeJSONHeaderValue)
 
-	var client *http.Client
-
+	clientOpts := []httpx.ResilientOptions{httpx.ResilientClientWithTracer(a.d.Tracer())}
 	if cfg.Api.Retry != nil {
-		maxRetryDelay := time.Second
-		giveUpAfter := time.Millisecond * 50
 		if len(cfg.Api.Retry.MaxDelay) > 0 {
+			maxRetryDelay := time.Second
```

Comment:
IMO this needs to be moved one scope up, like it was before, so that the default is also set when the config is not set.

Reply:
And we need a test for this, so e.g. only setting `cfg.Api.Retry.MaxDelay`, and expecting the `giveUpAfter` default to be set.

Reply:
After looking at this in more detail, the retry config was completely broken in https://github.com/ory/oathkeeper/commit/2a97e051a98da588aa8125bc0c6681e2d39c48ef.

I'll revisit this...

Reply:
Alrighty... the retry configuration is broken all over Oathkeeper. Here's my understanding of what is currently happening:

- The `give_up_after` property specifies the per-retry timeout. That's surprising to me, since it sounds like it means the overall timeout, including retries.
- The `max_delay` specifies the maximum backoff wait time in between retries.
- The maximum number of retries is not configurable, and is always 4 (the default in `hashicorp/retryablehttp`).
- This [config migration](https://github.com/ory/oathkeeper/blob/7150da04e91f515a391e95dd550d48066e25d841/rule/rule_migrator.go#L46-L88) is incorrect. It thinks `give_up_after` is the overall timeout (including retries).

Not sure now is the time to fix this fully, since that could also break users.

With the goal of moving this PR over the line, I propose to keep as much backward-compatibility as possible. This particular mutator previously had no overall timeout and no retries if the retry config was not specified at all. If the retry config was specified (even if `{}`), it peformed up to 4 retries. If the retry config is empty (`{}`), the timeouts also changed.

This patch keeps that behavior, while fixing bugs:
- the fallback timeouts (retry config present but empty) were wrong: 50ms timeout and 1s delay, even though the config schema promises a more reasonable 1s timeout and 100ms delay. Those defaults now at least work as advertised.
- faulty error messages

We should consider revamping the timeout handling all over Oathkeeper in a follow-up ticket.

Thoughts? @zepatrik @aeneasr @daviddelucca ?

Reply:
This PR is related to activate tracing on mutator. I think we could merge it and open another ticket/branch to work on revamping on timeouts.

Does make sense?